src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CreateClient.cs

```diff
@@ -51,6 +52,44 @@ public static void EmptyStringServerName_Throws_ArgumentException(PipeDirection
         }
 
         [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        [PlatformSpecific(PlatformID.Windows)]
+        public static void ReservedPipeName_Throws_ArgumentOutOfRangeException(PipeDirection direction)
+        {
+            const string serverName = ".";
+            const string reservedName = "anonymous";
+            Assert.Throws<ArgumentOutOfRangeException>("pipeName", () => new NamedPipeClientStream(reservedName));
+            Assert.Throws<ArgumentOutOfRangeException>("pipeName", () => new NamedPipeClientStream(serverName, reservedName));
+            Assert.Throws<ArgumentOutOfRangeException>("pipeName", () => new NamedPipeClientStream(serverName, reservedName, direction));
+            Assert.Throws<ArgumentOutOfRangeException>("pipeName", () => new NamedPipeClientStream(serverName, reservedName, direction, PipeOptions.None));
+            Assert.Throws<ArgumentOutOfRangeException>("pipeName", () => new NamedPipeClientStream(serverName, reservedName, direction, PipeOptions.None, TokenImpersonationLevel.Impersonation));
+        }
+
+        [DllImport("libc", SetLastError = true)]
+        private static extern unsafe int gethostname(byte* name, int len);
+
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public static void NotSupportedPipePath_Throws_PlatformNotSupportedException()
+        {
+            const int HOST_NAME_MAX = 255; // man gethostname
+            const int ArrLength = HOST_NAME_MAX + 1;
+            string hostName;
+
+            unsafe
+            {
+                byte* name = stackalloc byte[ArrLength];
+                Assert.True(gethostname(name, ArrLength) == 0, "gethostname failed");
```

Comment:
Why not Assert.Equal here?  This seems like a typical case where the message doesn't add additional value.